Maven properties in image name not replaced (Zero-Config mode)

### Describe the bug

In a Zero-Config mode it is currently not possible to use ${git.commit.id.abbrev} in the `jkube.generator.name` property or other properties defined by the git-commit-id-plugin. @rohanKanojia made the assumption that https://github.com/eclipse/jkube/blob/master/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/helper/ImageNameFormatter.java does not handle this. 


Here is the public demo project to reproduce this issue: https://github.com/kuhhpid/jkube-git-commit-id

### Eclipse JKube version

1.9.1

### Component

Kubernetes Maven Plugin

### Apache Maven version

_No response_

### Gradle version

_No response_

### Steps to reproduce

1. `git clone https://github.com/kuhhpid/jkube-git-commit-id`
2. `cd jkube-git-commit-id`
3. run `mvn package`
4. See described error that `${git.commit.id.abbrev}` was not replaced and violates the validation pattern 


### Expected behavior

The property `${git.commit.id.abbrev} ` should be replaced with the current git commit id abbrev.

### Runtime

other (please specify in additional context)

### Kubernetes API Server version

1.24.1

### Environment

Linux

### Eclipse JKube Logs

```shell
[INFO] ----------------< eu.awag.examples:jkube-git-commit-id >----------------
[INFO] Building jkube-git-commit-id 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- git-commit-id-plugin:4.9.10:revision (default) @ jkube-git-commit-id ---
[INFO] 
...
[INFO] 
[INFO] --- kubernetes-maven-plugin:1.9.1:resource (default) @ jkube-git-commit-id ---
[INFO] k8s: Running generator wildfly-jar
[INFO] k8s: wildfly-jar: Using Docker image quay.io/jkube/jkube-java:0.0.16 as base / builder
[INFO] k8s: Using resource templates from /home/alexander/IdeaProjects/jkube-git-commit-id/src/main/jkube
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.168 s
[INFO] Finished at: 2022-10-18T08:51:47+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.jkube:kubernetes-maven-plugin:1.9.1:resource (default) on project jkube-git-commit-id: Execution default of goal org.eclipse.jkube:kubernetes-maven-plugin:1.9.1:resource failed: Given Docker name 'registry.gitlab.com/myproject/myrepo/mycontainer:${git.commit.id.abbrev}' is invalid:
[ERROR]    * tag part '${git.commit.id.abbrev}' doesn't match allowed pattern '^[\w][\w.-]{0,127}$'
```


### Sample Reproducer Project

https://github.com/kuhhpid/jkube-git-commit-id

### Additional context

My Kubernates flavor is microk8s (https://microk8s.io)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -22,6 +22,7 @@ Usage:
 ```
 ### 1.11-SNAPSHOT
 * Fix #1546: Migrate to JUnit5 testing framework
+* Fix #1858: Properties in image name not replaced
 * Fix #1935: `oc:remote-dev` goal / `ocRemoteDev` task have wrong log prefixes
 
 ### 1.10.1 (2022-11-16)

--- a/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/helper/ImageNameFormatter.java
+++ b/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/helper/ImageNameFormatter.java
@@ -21,6 +21,9 @@ import java.util.Date;
 import java.util.HashMap;
 import java.util.Map;
 
+import static org.eclipse.jkube.kit.common.JKubeFileInterpolator.DEFAULT_FILTER;
+import static org.eclipse.jkube.kit.common.JKubeFileInterpolator.interpolate;
+
 /**
  * Replace placeholders in an image name with certain properties found in the
  * project
@@ -40,9 +43,11 @@ public class ImageNameFormatter implements ConfigHelper.NameFormatter {
     private final FormatParameterReplacer formatParamReplacer;
 
     private final Date now;
+    private JavaProject project;
 
     public ImageNameFormatter(JavaProject project, Date now) {
         this.now = now;
+        this.project = project;
         formatParamReplacer = new FormatParameterReplacer(initLookups(project));
     }
 
@@ -52,6 +57,7 @@ public class ImageNameFormatter implements ConfigHelper.NameFormatter {
             return null;
         }
 
+        name = interpolate(name, project.getProperties(), DEFAULT_FILTER);
         return formatParamReplacer.replace(name);
     }
 

--- a/jkube-kit/build/service/docker/src/test/java/org/eclipse/jkube/kit/build/service/docker/helper/ImageNameFormatterTest.java
+++ b/jkube-kit/build/service/docker/src/test/java/org/eclipse/jkube/kit/build/service/docker/helper/ImageNameFormatterTest.java
@@ -78,6 +78,8 @@ class ImageNameFormatterTest {
     @Test
     void artifact() {
         project.setArtifactId("Docker....Maven.....Plugin");
+        project.setProperties(new Properties());
+
         assertThat(formatter.format("--> %a <--")).isEqualTo("--> docker.maven.plugin <--");
     }
 
@@ -119,5 +121,15 @@ class ImageNameFormatterTest {
         final String result = formatter.format("%g/name");
         // Then
         assertThat(result).isEqualTo("this.it..is/name");
+    }
+
+    @Test
+    void format_whenPropertyInImageName_thenResolveProperty() {
+        // Given
+        project.getProperties().put("git.commit.id.abbrev", "der12");
+        // When
+        final String result = formatter.format("registry.gitlab.com/myproject/myrepo/mycontainer:${git.commit.id.abbrev}");
+        // Then
+        assertThat(result).isEqualTo("registry.gitlab.com/myproject/myrepo/mycontainer:der12");
     }
 }

